src/index.js

```diff
@@ -337,58 +339,88 @@ function Bigtable(options) {
 
   options = common.util.normalizeArguments(this, options);
 
-  var baseUrl = 'bigtable.googleapis.com';
-  var adminBaseUrl = 'bigtableadmin.googleapis.com';
+  // Determine what scopes are needed.
+  // It is the union of the scopes on all three clients.
+  let scopes = [];
+  let clientClasses = [
+    v2.BigtableClient,
+    v2.BigtableInstanceAdminClient,
+    v2.BigtableTableAdminClient,
+  ];
+  for (let clientClass of clientClasses) {
+    for (let scope of clientClass.scopes) {
+      if (scopes.indexOf(scope) === -1) {
+        scopes.push(scope);
+      }
+    }
+  }
+
+  var defaultBaseUrl = 'bigtable.googleapis.com';
+  var defaultAdminBaseUrl = 'bigtableadmin.googleapis.com';
 
   var customEndpoint =
     options.apiEndpoint || process.env.BIGTABLE_EMULATOR_HOST;
+  var customEndpointBaseUrl;
+  var customEndpointPort;
 
   if (customEndpoint) {
-    baseUrl = customEndpoint;
-    adminBaseUrl = baseUrl;
+    var customEndpointParts = customEndpoint.split(':');
+    customEndpointBaseUrl = customEndpointParts[0];
+    customEndpointPort = customEndpointParts[1];
   }
 
-  var config = {
-    baseUrl: baseUrl,
-    customEndpoint: !!customEndpoint,
-    protosDir: path.resolve(__dirname, '../protos'),
-    protoServices: {
-      Bigtable: {
-        baseUrl: baseUrl,
-        path: 'google/bigtable/v2/bigtable.proto',
-        service: 'bigtable.v2',
-      },
-      BigtableTableAdmin: {
-        baseUrl: adminBaseUrl,
-        path: 'google/bigtable/admin/v2/bigtable_table_admin.proto',
-        service: 'bigtable.admin.v2',
+  var options_ = extend(
```

Comment:
This should probably have better naming, for example userOptions, defaultOptions, and options